Create title component

Bots opened the pull request:
- Created a simple title component

Bots committed:
chore: added title component

Bots committed:
chore: removed package lock

RedBlueKoi commented on src/components/TitleItem.vue:
```diff
@@ -0,0 +1,20 @@
+</template>
+
+<script setup lang="ts">
+  const props = defineProps(["title"])
```
If you are not planning to use your props in the `script` section then you don't have to assign them to a variable, since they are gonna be exposed automatically.

RedBlueKoi replied:
We are using TypeScript. So please define props types using interface. 
Here is an example:
```vue
<script setup lang="ts">
interface Props {
  foo: string
  bar?: number
}

const props = defineProps<Props>()
</script>
```
If you are not comfortable with TS or have any questions feel free to ping me in chat or in private and we can figure it out

RedBlueKoi commented on src/components/TitleItem.vue:
```diff
@@ -0,0 +1,20 @@
+<template>
```
File name should be representative of its purpose. Personally I would say that PageTitle is more fitting for this component.

RedBlueKoi commented on src/components/TitleItem.vue:
```diff
@@ -0,0 +1,20 @@
+<template>
+  <h1>
+    {{ props.title }}
```
As I said, props are exposed automatically, so you can access them straight away.
```vue
<h1>{{ title }}</h1>
```

RedBlueKoi commented on src/components/TitleItem.vue:
```diff
@@ -0,0 +1,20 @@
+</script>
+
+<style scoped lang="postcss">
+  h1 {
```
Alright, a couple of things to mention here:
1. In general, try to avoid using tags as selectors. that's a bad practice because of multiple reasons. If you want to know more why, feel free to ping me.
2. We are using Tailwindcss in the project and literally everything you've done here can be done in Tailwind. So please use that. If you have any questions regarding Tailwind, you can check out the documentation in the README of the project or just ping me.
3. Hardcoded color should be also moved in the Tailwind and used as a class, or, at the very least, used as a custom class ```color-[#34495e]```

RedBlueKoi commented on src/views/HomeView.vue:
```diff
@@ -1,5 +1,9 @@
 <template>
-  <main></main>
+  <main>
+    <TitleItem title="Title" />
```
By Vue convention it is preferred to use kebab-case when using components in the template: 
```<title-item />```

RedBlueKoi reviewed (changes requested):
Left some comments. I know there are a bit too many, but the code in general looks good, I feel like you can make it a bit more polished 🙂 It was not an attempt to overwhelm you. If you have any questions feel free to ping me and we can have a chat

Bots committed:
chore: fixed issues found in review

- Defined interface for props
- Changed filename
- Used tailwinds for styling

RedBlueKoi commented on src/components/PageTitle.vue:
```diff
@@ -0,0 +1,15 @@
+<template>
+  <h1
+    class="text-[40px] font-[400] font-normal leading-[48px] color-[#34495e] not-italic">
```
One small nit-pick here:
`font-[400] font-normal not-italic` are not required here since this is a top level component. They are not gonna be affected unless, let's say, we make the entire page italic, in which case I suppose the title should be italic too 🙂 
Also, you can combine `font-size` and `line-height` by using Tailwind typography - https://tailwindcss.com/docs/font-size 
I know that the page title is 40px in the wireframe, but I would suggest making what makes sense instead of doing pixel-perfect. If default config is close enough then it's good enough. Don't make your life more difficult 🙂

RedBlueKoi replied:
Color should be moved to tailwind config, but ignore it for now, since it's gonna be done in a different issue

Bots committed:
chore: moved styling to tailwind config

- Moved title text color to tailwind config
- Moved title textsize and line height to tailwind config
- Removed unnecessary styling

RedBlueKoi committed:
feat: small adjustments, added default colors in tailwind config
